spec/index.adoc

```diff
@@ -4264,6 +4265,8 @@ query_methods[method] = λ (arg, caller, sysenv) → λ wasm_state →
   if es.calls ≠ [] then Trap {cycles_used = es.cycles_used;}
   if es.ingress_filter ≠ Reject then Trap {cycles_used = es.cycles_used;}
   if es.response = NoResponse then Trap {cycles_used = es.cycles_used;}
+  if es.new_certified_data ≠ NoCertifiedData then Trap {cycles_used = es.cycles_used;}
+  if es.new_global_timer ≠ NoGlobalTimer then Trap {cycles_used = es.cycles_used;}
```

Comment:
Maybe this pattern for error handling doesn’t quite scale anymore :-)

Maybe it is more direct to have an `es.context` that is an enumeration of all the various contexts for system calls (like that table in the spec), and the assert the right context in each system call.

Or remove all that noise from here and _just_ point to the table, i.e. make the trap implicit.

Reply:
I've had this on my TODO list and your comment made me implement this. Thank you!